src/main/java/com/firebolt/jdbc/exception/ServerError.java

```diff
@@ -29,12 +28,23 @@ public ServerError(Query query, Error[] errors) {
     }
 
     public ServerError(JSONObject json) {
-        this(fromJson(json.optJSONObject("query"), Query::new), fromJson(json.optJSONArray("errors"), Error::new, Error[]::new));
+        JSONArray errorArray =  json.optJSONArray("errors");
+        this.query = fromJson(json.optJSONObject("query"), Query::new);
+        if(errorArray == null) {
+            this.errors = new Error[] {new Error(json, false)};
+        } else {
+            this.errors = fromJson(json.optJSONArray("errors"));
+        }
     }
 
-    private static <T> T[] fromJson(JSONArray jsonArray, Function<JSONObject, T> factory, IntFunction<T[]> arrayFactory) {
-        return jsonArray == null ? null : IntStream.range(0, jsonArray.length()).boxed().map(jsonArray::getJSONObject).map(factory).toArray(arrayFactory);
-    }
+	private static Error[] fromJson(JSONArray jsonArray) {
+		return jsonArray == null
+                ? null
+				: IntStream.range(0, jsonArray.length()).boxed()
+                        .map(jsonArray::getJSONObject)
+                        .map(jsonObject -> new Error(jsonObject, true))
+						.toArray(Error[]::new);
+	}
```

Comment:
There are some tabs here as well. Let's clean them up so we have everything following the same standard. Please check the other files too.